rust-symcrypt/src/rsa.rs

```diff
@@ -0,0 +1,643 @@
+/// !REVIEW: I've tried to closely follow SymCrypts API calls and semantics making some simplifications, but I think we have a chance to simplify further if we choose to.
+/// My questions / Suggestions:
+/// 1. We've introduced the idea of the RsaKeyPair / RsaPublicKey state, which can hold a lot of data, and makes a lot of the calls to the underlying SymCrypt code redundant. most of the accessors no longer
+/// needed to call SymCrypt since the same data is stored on the state or is assumed
```

Comment:
This doesn't seem to be true in the latest iteration? I'd prefer to keep all the state in the SymCrypt object. We don't want it duplicated in the Rust layer because then there's the possibility of things getting out-of-sync.